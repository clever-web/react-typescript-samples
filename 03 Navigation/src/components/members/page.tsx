import * as React from 'react';

export const MembersPage: React.StatelessComponent<{}> = () => {
  return (
    <div className="row">
      <h2> Members Page</h2>
    </div>
  );
}
