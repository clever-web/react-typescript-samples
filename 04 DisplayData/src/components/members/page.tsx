import * as React from 'react';
import { MemberEntity } from '../../model';
import { memberAPI } from '../../api/member';

interface State {
  members: MemberEntity[];
}

export class MembersPage extends React.Component<Props, State> {

  constructor() {
    super();
    this.state = { members: [] };
  }

  public componentWillMount() {
    this.state.members = memberAPI.fetchMembers();
  }

  public render() {
    var CreateMemberRow = function (member: MemberEntity) {
      return (
        <tr key={member.id}>
          <td>
            <img src={member.avatar_url} className="avatar" />
          </td>
          <td>
            <span>{member.id}</span>
          </td>
          <td>
            <span>{member.login}</span>
          </td>
        </tr>
      )
    }

    return (
      <div className="row">
        <h2> Members Page</h2>
        <table className="table">
          <thead>
            <tr>
              <th>
                Avatar
                </th>
              <th>
                Id
                </th>
              <th>
                Name
                </th>
            </tr>
          </thead>
          <tbody>
            {this.state.members.map(CreateMemberRow, this)}
          </tbody>
        </table>
      </div>
    );
  }
}
