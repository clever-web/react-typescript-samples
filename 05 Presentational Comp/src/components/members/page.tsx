import * as React from 'react';
import { MemberEntity } from '../../model';
import { memberAPI } from '../../api/member';
import { MemberHeader } from './memberHeader';
import { MemberRow } from './memberRow';

interface State {
  members: MemberEntity[];
}

export class MembersPage extends React.Component<{}, State> {
  constructor() {
    super();
    this.state = { members: [] };
  }

  public componentDidMount() {
    memberAPI.fetchMembers()
      .then((members) => {
        this.setState({ members });
      });
  }

  public render() {
    return (
      <div className="row">
        <h2> Members Page</h2>
        <table className="table">
          <thead>
            <MemberHeader />
          </thead>
          <tbody>
            {
              this.state.members.map((member) =>
                <MemberRow
                  key={member.id}
                  member={member}
                />
              )
            }
          </tbody>
        </table>
      </div>
    );
  }
};
