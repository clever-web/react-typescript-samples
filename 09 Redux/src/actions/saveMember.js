var memberAPI_1 = require('../api/memberAPI');
var memberFormValidator_1 = require('../validations/memberFormValidator');
var saveMember = function (member) {
    var errorsSave = memberFormValidator_1.default.validateMember(member);
    if (errorsSave.isEntityValid) {
        memberAPI_1.default.saveAuthor(member);
    }
    return {
        type: 'MEMBER_SAVE',
        errors: errorsSave
    };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = saveMember;
//# sourceMappingURL=saveMember.js.map