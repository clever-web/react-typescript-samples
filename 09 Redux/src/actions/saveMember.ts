import MemberEntity from "../api/MemberEntity"

const saveMember = () => {
   return {
     type: 'MEMBER_SAVE'
   }
}

export default saveMember;
