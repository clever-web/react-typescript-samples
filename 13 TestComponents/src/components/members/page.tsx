import * as React from 'react';
import { Link } from 'react-router-dom';
import { MemberEntity } from '../../model';
import { MemberHeader } from './memberHeader';
import { MemberRow } from './memberRow';

export interface Props {
  members: MemberEntity[];
  fetchMembers(): void;
}

export class MembersPage extends React.Component<Props,{}> {
  constructor(props) {
    super(props);
    this.state = { members: [] };
  }
  public componentDidMount() {
    this.props.fetchMembers();
  }

  public render() {
    return (
      <div className="row">
        <h2> Members Page</h2>
        <Link to="/member">New Member</Link>
        <table className="table">
          <thead>
            <MemberHeader />
          </thead>
          <tbody>
            {
              this.props.members.map((member) =>
                <MemberRow
                  key={member.id}
                  member={member}
                />
              )
            }
          </tbody>
        </table>
      </div>
    );
  }
};
