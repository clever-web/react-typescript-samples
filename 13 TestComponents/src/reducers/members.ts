import { actionTypes } from '../common/constants/actionTypes';
import { MemberEntity } from '../model';


export const createEmptyMembers = (): MemberEntity[] =>(
  []
)
export const membersReducer = (state: MemberEntity[] = [], action) => {
  switch (action.type) {
    case actionTypes.FETCH_MEMBERS_COMPLETED:
      return handleFetchMembersCompleted(state, action.payload);
  }

  return state;
};

const handleFetchMembersCompleted = (state: MemberEntity[], payload: MemberEntity[]) => {
  return payload;
};
