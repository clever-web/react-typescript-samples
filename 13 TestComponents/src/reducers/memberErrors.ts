import { FieldValidationResult, FormValidationResult } from 'lc-form-validation';
import { actionTypes } from '../common/constants/actionTypes';
import { MemberErrors } from '../model';
import { MemberFieldChangePayload } from '../components/member/actions/memberFieldChange';

export const createEmptyMemberErrors = (): MemberErrors => ({
  login: new FieldValidationResult(),
});

export const memberErrorsReducer = (state = createEmptyMemberErrors(), action) => {
  switch (action.type) {
    case actionTypes.FETCH_MEMBERS_COMPLETED:
      return handleFetchMembersCompleted(state, action.payload);
    case actionTypes.UPDATE_MEMBER_FIELD:
      return handleUpdateMemberField(state, action.payload);
    case actionTypes.SAVE_MEMBER:
      return handleSaveMember(state, action.payload);
  }

  return state;
};

const handleFetchMembersCompleted = (state: MemberErrors, payload) => {
  return createEmptyMemberErrors();
};

const handleUpdateMemberField = (state: MemberErrors, payload: MemberFieldChangePayload): MemberErrors => {
  return {
    ...state,
    [payload.fieldValidationResult.key]: payload.fieldValidationResult,
  };
};

const handleSaveMember = (state: MemberErrors, payload: FormValidationResult): MemberErrors => {
  const newMemberErrors = { ...state };

  return payload.fieldErrors.reduce((memberErrors, fieldValidationResult) => {
    memberErrors[fieldValidationResult.key] = fieldValidationResult;
    return memberErrors;
  }, newMemberErrors);
};
