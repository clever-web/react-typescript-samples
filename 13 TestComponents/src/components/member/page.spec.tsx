import * as React from 'react';
import {MemberPage, Props} from './page';
import * as SaveMember from './actions/saveMember';
import {shallow} from 'enzyme';
import { createEmptyMember } from '../../reducers/member';
import { createEmptyMemberErrors } from '../../reducers/memberErrors';
import { MemberEntity } from '../../model';


describe ('/components/member/page specs',()=>{
    it('should return as expected when props are passed',()=>{
        //arrange
        const props:Props = {
            memberId:-1,
            member: createEmptyMember(),
            memberErrors: createEmptyMemberErrors(),
            fetchMemberById: () => {},
            onChange: () =>{},
            onSave: ()=>{}
        }
        //act
        const component = shallow (
            <MemberPage {...props} />
        );
        //assert
        expect(component).toMatchSnapshot();
    });

    it('should call onSave when onSave is activated',()=>{
        //arrange
        const newMember:MemberEntity = {
            id:56789,
            avatar_url:'test new avatar',
            login: 'test new login'
        }
        const props: Props = {
            memberId: -1,
            member: newMember,
            memberErrors: createEmptyMemberErrors(),
            fetchMemberById: () => { },
            onChange: () => { },
            onSave: jest.fn(),
        }
       

        const actionSaveStub = jest.spyOn(SaveMember, 'saveMemberAction')
             .mockImplementation(()=>(
               { ...newMember,}
            ));
               
        const component = shallow(
            <MemberPage {...props} />,
        );

        
        //component.prop('onSave')();
        component.simulate('save',newMember);
       
        //assert
        expect(props.onSave).toHaveBeenCalledWith(newMember);

    });

    it('should call onChange when Changes are made', () => {
        //arrange

         const newMember: MemberEntity = {
             id: 56789,
             avatar_url: 'test new avatar',
             login: 'test new login'
         }
        const props: Props = {
            memberId: -1,
            member: newMember,
            memberErrors: createEmptyMemberErrors(),
            fetchMemberById: () => { },
            onChange: jest.fn(),
            onSave: () => { } ,
        }

        let fieldName =  'test fieldName';
        let value =  'test value';

        //act
        const component = shallow(
            <MemberPage {...props} />
        );


         component.simulate('change', fieldName, value);
        //assert
         expect(props.onChange).toHaveBeenCalledWith(newMember,fieldName, value);

    }); 
    it ('should componetDidMount when component is called',()=>{
        //arrange
        const props: Props = {
            memberId: -1,
            member: createEmptyMember(),
            memberErrors: createEmptyMemberErrors(),
            fetchMemberById: jest.fn(),
            onChange: ()=>{},
            onSave: () => { },
        }

        //act
        const component = shallow(
            <MemberPage {...props} />
        );

        component.find('componentDidMount');
       
        //assert
        expect (props.fetchMemberById).toHaveBeenCalled();
    });
});