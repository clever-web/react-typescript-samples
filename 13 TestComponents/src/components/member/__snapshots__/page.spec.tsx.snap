// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/components/member/page specs should return as expected when props are passed 1`] = `
<Component
  member={
    Object {
      "avatar_url": "",
      "id": -1,
      "login": "",
    }
  }
  memberErrors={
    Object {
      "login": FieldValidationResult {
        "errorMessage": "",
        "key": "",
        "succeeded": false,
        "type": "",
      },
    }
  }
  onChange={[Function]}
  onSave={[Function]}
/>
`;
